src/app/test/service/testRunner.ts

```diff
@@ -24,10 +28,13 @@ export async function runTests(testSession: TestSession) {
 
   const testCases = await loadTestCases()
 
-  for (const testCase of testCases) {
-    const result = await runTest(testSession, testCase)
-    testSession.addResult(result)
-  }
+  await Promise.all(
```

Comment:
`.map` creates an array of `undefined`s and this array is passed to the `Promise.all`. This seems like a misuse of either `.map` or `Promise.all`, or both.

Reply:
Not really, it maps to an `Promise<void>[]` which then is used by `Promise.all` to run these promises in parallel

Reply:
We've resolved this over a huddle.